### PR TITLE
Improve net construction

### PR DESCRIPTION
**Pull request by aidangomez**
- [x] Concatenation
  
  `[L1, L2] => L3`
- [x] Splitting
  
  `L1.fromOffset(0) => L2`
  `L1.fromOffset(10) => L3`
- [x] Linear connection
  
  `L1 => L2 => L3`

### CODE REVIEW

**Comment by aidangomez** [comment]
@aleph7 This has been updated to what we talked about yesterday

File: Source/Net.swift [hunk 1]
```diff
@@ -17,6 +17,14 @@ public class Net {
     var sinkNodes = [NetNode]()
     var nodes = [NetNode]()
 
+    static internal var buildNet = Net()
```
Comment (alejandro-isaza) [inline Source/Net.swift]: Let use a stack here. Nested calls to `Net.build` are unlikely but at least it won't give unexpected results if someone ever does it.

File: Source/NetUtilities.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import Metal
+
+
+
+infix operator => { associativity left precedence 160 }
+public func =>(lhs: Layer, rhs: Layer) -> Layer {
+    let net = Net.buildNet
+
+    let lhsLayer: Net.LayerRef
+    if let node = net.nodeWithLayerName(lhs.name) {
```
Comment (alejandro-isaza) [inline Source/NetUtilities.swift]: Let's use `NSUUID` instead

File: Source/NetUtilities.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        rhsRef = net.addLayer(rhs)
+    }
+
+    net.connectBuffer(lhs.bufferRef, atOffset: lhs.offset, toLayer: rhsRef)
+
+    return rhs
+}
+
+public extension Layer {
+    public subscript(offset: Int) -> (bufferRef: Net.BufferRef, offset: Int) {
```
Comment (alejandro-isaza) [inline Source/NetUtilities.swift]: change this to `public func fromOffset(offset: Int) ...`

Comment (alejandro-isaza) [inline Source/NetUtilities.swift]: Also add a documentation block explaining what this does and why it doesn't take a size.

**Pull request merged**